tests/actor/teststatemanager.py

```diff
@@ -0,0 +1,308 @@
+
+        val = await state_manager.add_or_update_state('state1', 'value1', test_update_value)
+        self.assertEqual('value1', val)
+
+    async def test_add_or_update_state_for_none_state_key(self):
+        """update state value for StateChangeKind.none state """
+        def test_update_value(name, value):
+            return f'{name}-{value}'
+
+        self._fake_client.get_state.return_value = base64.b64encode(b'"value1"')
+        state_manager = ActorStateManager(self._fake_actor)
+        has_value, val = await state_manager.try_get_state('state1')
+        self.assertTrue(has_value)
+        self.assertEqual('value1', val)
+
+        val = await state_manager.add_or_update_state('state1', 'value1', test_update_value)
+        self.assertEqual('state1-value1', val)
+    
+    async def test_add_or_update_state_without_update_value_factory(self):
+        """tries to add or update state without update_value_factory """
+        state_manager = ActorStateManager(self._fake_actor)
+        with self.assertRaises(AttributeError):
+            await state_manager.add_or_update_state('state1', 'value1', None)
+
+    async def test_get_state_names(self):
+        state_manager = ActorStateManager(self._fake_actor)
+        await state_manager.set_state('state1', 'value1')
+        await state_manager.set_state('state2', 'value2')
+        await state_manager.set_state('state3', 'value3')
+        names = await state_manager.get_state_names()
+        self.assertEqual(['state1', 'state2', 'state3'], names)
+
+    async def test_clear_cache(self):
+        state_manager = ActorStateManager(self._fake_actor)
+        await state_manager.set_state('state1', 'value1')
+        await state_manager.set_state('state2', 'value2')
+        await state_manager.set_state('state3', 'value3')
+        await state_manager.clear_cache()
+
+        self.assertEqual(0, len(state_manager._state_change_tracker))
+
+    async def test_save_state(self):
+        state_manager = ActorStateManager(self._fake_actor)
+        # set states which are StateChangeKind.add
+        await state_manager.set_state('state1', 'value1')
+        await state_manager.set_state('state2', 'value2')
+        # set state which is StateChangeKind.none
+        self._fake_client.get_state.return_value = base64.b64encode(b'"value3"')
+        has_value, val = await state_manager.try_get_state('state3')
+        self.assertTrue(has_value)
+        self.assertEqual("value3", val)
+        # set state which is StateChangeKind.remove
+        await state_manager.remove_state('state4')
+        # set state which is StateChangeKind.update
+        await state_manager.set_state('state5', 'value5')
+
+        expected = b'[{"operation":"upsert","request":{"key":"state1","value":"InZhbHVlMSI="}},{"operation":"upsert","request":{"key":"state2","value":"InZhbHVlMiI="}},{"operation":"delete","request":{"key":"state4"}},{"operation":"upsert","request":{"key":"state5","value":"InZhbHVlNSI="}}]'
```

Comment:
@artursouza this is json formatted data which will be stored in actor state store. dotnet one always encoded value in base64... please review this expected format.

cc/ @amanbha

Reply:
I know .Net does that. Java does not. I remember @amanbha talking about changing this to be simply JSON and not base64 encoded anymore.

Reply:
@amanbha @artursouza  I can still see that dotnet-sdk encodes value to base64. What's the decision?

Reply:
My understanding is that we will save it as JSON. .Net simply has not changed that yet. @amanbha ?

Reply:
If you do not have any concerns for the other implementation, I would like to merge this PR and revisit serialization later before we merge feature branch to master. 

I want to bring timer/reminder PR up now and demo python actor sdk in the next week community call.

Reply:
By default C# will serialize as Json, but when serializer is overridden it will send as base64

Reply:
oh.. :( my cloned repo that I am referring to is really old... I found that dotnet uses the default json serializer. 

https://github.com/dapr/dotnet-sdk/blob/f138d75192f68d9a1f35986ebdeae0ef3c623764/src/Dapr.Actors/Runtime/DaprStateProvider.cs#L107-L115

I created https://github.com/dapr/python-sdk/issues/38 issue to revisit this problem.

Reply:
@artursouza @amanbha Please approve if you do not have any other concerns. I will revisit serialization later.